packages/compiler/test/checker/templates.test.ts

```diff
@@ -515,4 +515,230 @@ describe("compiler: templates", () => {
+
+    it("with named arguments out of order", async () => {
+      testHost.addTypeSpecFile(
+        "main.tsp",
+        `
+          model A<T, U> { a: T, b: U }
+          @test model B {
+            foo: A<U = int32, T = string>
+          };
+        `
+      );
+
+      const { B } = (await testHost.compile("main.tsp")) as { B: Model };
+      const foo = B.properties.get("foo")!.type;
+      strictEqual(foo.kind, "Model");
+      const a = foo.properties.get("a")!;
+      const b = foo.properties.get("b")!;
+      strictEqual(a.type.kind, "Scalar");
+      strictEqual(a.type.name, "string");
+      strictEqual(b.type.kind, "Scalar");
+      strictEqual(b.type.name, "int32");
+    });
+
+    it("with named arguments and defaults", async () => {
+      testHost.addTypeSpecFile(
+        "main.tsp",
+        `
+          model A<T = int32, U = string> { a: T, b: U }
+          @test model B {
+            foo: A<U = "bar">
+          };
+        `
+      );
+
+      const { B } = (await testHost.compile("main.tsp")) as { B: Model };
+      const foo = B.properties.get("foo")!.type;
+      strictEqual(foo.kind, "Model");
+      const a = foo.properties.get("a")!;
+      const b = foo.properties.get("b")!;
+      strictEqual(a.type.kind, "Scalar");
+      strictEqual(a.type.name, "int32");
+      strictEqual(b.type.kind, "String");
+      strictEqual(b.type.value, "bar");
+    });
+
+    it("with named and positional arguments", async () => {
+      testHost.addTypeSpecFile(
+        "main.tsp",
+        `
+          model A<T, U = int32, V extends string = string> { a: T, b: U, c: V }
+
+          @test model B {
+            foo: A<boolean, V = "bar">
+          }
+
+          @test model C {
+            foo: A<T = boolean, V = "bar">
+          }
+        `
+      );
+
+      const { B, C } = (await testHost.compile("main.tsp")) as { B: Model; C: Model };
+
+      for (const M of [B, C]) {
+        const foo = M.properties.get("foo")!.type;
+        strictEqual(foo.kind, "Model");
+        const a = foo.properties.get("a")!;
+        const b = foo.properties.get("b")!;
+        const c = foo.properties.get("c")!;
+        strictEqual(a.type.kind, "Scalar");
+        strictEqual(a.type.name, "boolean");
+        strictEqual(b.type.kind, "Scalar");
+        strictEqual(b.type.name, "int32");
+        strictEqual(c.type.kind, "String");
+        strictEqual(c.type.value, "bar");
+      }
+    });
+
+    it("cannot specify positional argument after named argument", async () => {
+      const { pos, end, source } = extractSquiggles(`
+      model A<T, U, V extends string = string> { a: T, b: U, c: V }
+
+      @test model B {
+        foo: ~~~A<boolean, V = "bar", string>~~~
+      }
+    `);
+
+      testHost.addTypeSpecFile("main.tsp", source);
+
+      const [{ B }, diagnostics] = (await testHost.compileAndDiagnose("main.tsp")) as [
+        { B: Model },
+        Diagnostic[],
+      ];
+
+      const foo = B.properties.get("foo")!.type;
```

Comment:
nit: does it matter to test the content of this template if there was an error creating it?

Reply:
This test is just making sure I didn't break anything in the logic about how to instantiate unspecified types. We instantiate them to their constraint (or unknown in this case as the implied constraint of an unconstrained parameter).